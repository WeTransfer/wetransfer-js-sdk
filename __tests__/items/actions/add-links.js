const routes = require('../../../src/config/routes');
const addLinksAction = require('../../../src/items/actions/add-links');
const { futureLink, RemoteLink } = require('../../../src/items/models');
const { RemoteTransfer } = require('../../../src/transfer/models');

describe('Add links action', () => {
  let addLinks = null;
  let transfer = null;
  const mocks = {};
  beforeEach(() => {
    mocks.request = { send: jest.fn() };
    mocks.request.send.mockReturnValue([
      {
        id: 'random-hash',
        content_identifier: 'web_content',
        meta: {
          title: 'WeTransfer'
        },
        url: 'https://wetransfer.com'
      }
    ]);

    addLinks = addLinksAction({
      routes,
      request: mocks.request,
      futureLink,
      RemoteLink
    });

    transfer = new RemoteTransfer();
  });

  it('should create an add items request', async () => {
    const linkItems = await addLinks(transfer, [
      {
        url: 'https://wetransfer.com',
        meta: {
          title: 'WeTransfer'
        }
      }
    ]);
    expect(linkItems).toMatchSnapshot();
  });
});
