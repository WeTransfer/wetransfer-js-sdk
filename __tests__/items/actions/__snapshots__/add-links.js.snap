// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add links action should create an add items request 1`] = `
Array [
  RemoteLink {
    "content_identifier": "web_content",
    "id": "random-hash",
    "meta": Object {
      "title": "WeTransfer",
    },
    "transfer": RemoteTransfer {
      "items": Array [
        [Circular],
      ],
    },
  },
]
`;
